File: 443-string-compression/string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        vector<char> ans;
        for(int i=0;i<chars.size();){
            char t = chars[i];
            int count=0;
            while(i<chars.size() && chars[i]==t){
                count++;
                i++;
            }
            ans.push_back(t);
            if(count==1){
                continue;
            }
            else{
                string u = to_string(count);
                for(int j=0;j<u.size();j++){
                    ans.push_back(u[j]);
                }
            }
        }
        chars = ans;
        return chars.size();
    }
};

File: 443-string-compression/string-compression.cs
public class Solution {
    public int Compress(char[] chars) {
        int n = chars.Length;
        String com = "";
        for(int i=0;i<n;i++){
            char t = chars[i];
            int count=1;
            while(i+1<n && chars[i+1]==t){
                count++;
                i+=1;
            }
            com+=t;
            if(count>1){
                com+=count.ToString();
            }
        }
        // chars = com.ToCharArray();
        Array.Copy(com.ToCharArray(),chars,com.Length);
        Array.Resize(ref chars,com.Length);
        return chars.Length;
    }
}